index.d.ts

```diff
@@ -194,6 +194,25 @@ declare module "replicate" {
         deployment_owner: string,
         deployment_name: string
       ): Promise<Deployment>;
+      create(deployment_config: {
+        name: string;
+        model: string;
+        version: string;
+        hardware: string;
+        min_instances: number;
+        max_instances: number;
+      }): Promise<Deployment>;
+      update(
+        deployment_owner: string,
+        deployment_name: string,
+        deployment_config: {
+          version: string;
+          hardware: string;
+          min_instances: number;
+          max_instances: number;
+        }
+      ): Promise<Deployment>;
```

Comment:
Make config fields optional but require at least one to be specified.

```suggestion
      update(
        deployment_owner: string,
        deployment_name: string,
        deployment_config: {
          version?: string;
          hardware?: string;
          min_instances?: number;
          max_instances?: number;
        } & (
          | { version: string }
          | { hardware: string }
          | { min_instances: number }
          | { max_instances: number }
        )
      ): Promise<Deployment>;
```

Reply:
Not at the moment. All fields are required on the API side for creation at the moment, but something we should fix.

Reply:
Sorry, misread this, yes this is correct. A good fix 👍